```
2021-11-26T20:41:34.1698874Z =================================== FAILURES ===================================
2021-11-26T20:41:34.1704846Z ______________________ TestPageBuffering.test_actual_pbs _______________________
2021-11-26T20:41:34.1710442Z 
2021-11-26T20:41:34.1716237Z self = <h5py.tests.test_file.TestPageBuffering testMethod=test_actual_pbs>
2021-11-26T20:41:34.1721714Z 
2021-11-26T20:41:34.1736939Z     def test_actual_pbs(self):
2021-11-26T20:41:34.1737538Z         """Verify actual page buffer size."""
2021-11-26T20:41:34.1737952Z         fname = self.mktemp()
2021-11-26T20:41:34.1738323Z         fsp = 16 * 1024
2021-11-26T20:41:34.1738664Z         pbs = 2 * fsp
2021-11-26T20:41:34.1739753Z         with File(fname, mode='w', fs_strategy='page', fs_page_size=fsp):
2021-11-26T20:41:34.1740228Z             pass
2021-11-26T20:41:34.1740846Z >       with File(fname, mode='r', page_buf_size=pbs-1) as f:
2021-11-26T20:41:34.1741169Z 
2021-11-26T20:41:34.1742163Z ../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.8/site-packages/h5py/tests/test_file.py:235: 
2021-11-26T20:41:34.1742962Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-11-26T20:41:34.1744054Z ../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.8/site-packages/h5py/_hl/files.py:507: in __init__
2021-11-26T20:41:34.1745033Z     fid = make_fid(name, mode, userblock_size, fapl, fcpl, swmr=swmr)
2021-11-26T20:41:34.1746136Z ../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.8/site-packages/h5py/_hl/files.py:220: in make_fid
2021-11-26T20:41:34.1746885Z     fid = h5f.open(name, flags, fapl=fapl)
2021-11-26T20:41:34.1747355Z h5py/_objects.pyx:54: in h5py._objects.with_phil.wrapper
2021-11-26T20:41:34.1747735Z     ???
2021-11-26T20:41:34.1748123Z h5py/_objects.pyx:55: in h5py._objects.with_phil.wrapper
2021-11-26T20:41:34.1748492Z     ???
2021-11-26T20:41:34.1748899Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-11-26T20:41:34.1749199Z 
2021-11-26T20:41:34.1749502Z >   ???
2021-11-26T20:41:34.1749894Z E   OSError: Unable to open file (page buffering is disabled for parallel)
2021-11-26T20:41:34.1750203Z 
2021-11-26T20:41:34.1750517Z h5py/h5f.pyx:106: OSError
2021-11-26T20:41:34.1750956Z __________________ TestPageBuffering.test_check_page_buf_size __________________
2021-11-26T20:41:34.1751254Z 
2021-11-26T20:41:34.1751652Z self = <h5py.tests.test_file.TestPageBuffering testMethod=test_check_page_buf_size>
2021-11-26T20:41:34.1751950Z 
2021-11-26T20:41:34.1752310Z     def test_check_page_buf_size(self):
2021-11-26T20:41:34.1752774Z         """Verify set page buffer size, and minimum meta and raw eviction criteria."""
2021-11-26T20:41:34.1753220Z         fname = self.mktemp()
2021-11-26T20:41:34.1753566Z         pbs = 16 * 1024
2021-11-26T20:41:34.1753909Z         mm = 19
2021-11-26T20:41:34.1754224Z         mr = 67
2021-11-26T20:41:34.1754779Z >       with File(fname, mode='w', fs_strategy='page',
2021-11-26T20:41:34.1755304Z                   page_buf_size=pbs, min_meta_keep=mm, min_raw_keep=mr) as f:
2021-11-26T20:41:34.1755634Z 
2021-11-26T20:41:34.1756594Z ../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.8/site-packages/h5py/tests/test_file.py:214: 
2021-11-26T20:41:34.1757533Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-11-26T20:41:34.1758638Z ../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.8/site-packages/h5py/_hl/files.py:507: in __init__
2021-11-26T20:41:34.1759411Z     fid = make_fid(name, mode, userblock_size, fapl, fcpl, swmr=swmr)
2021-11-26T20:41:34.1760490Z ../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.8/site-packages/h5py/_hl/files.py:226: in make_fid
2021-11-26T20:41:34.1761261Z     fid = h5f.create(name, h5f.ACC_TRUNC, fapl=fapl, fcpl=fcpl)
2021-11-26T20:41:34.1761736Z h5py/_objects.pyx:54: in h5py._objects.with_phil.wrapper
2021-11-26T20:41:34.1762125Z     ???
2021-11-26T20:41:34.1762490Z h5py/_objects.pyx:55: in h5py._objects.with_phil.wrapper
2021-11-26T20:41:34.1763010Z     ???
2021-11-26T20:41:34.1763447Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-11-26T20:41:34.1763758Z 
2021-11-26T20:41:34.1764041Z >   ???
2021-11-26T20:41:34.1764451Z E   OSError: Unable to create file (page buffering is disabled for parallel)
2021-11-26T20:41:34.1764746Z 
2021-11-26T20:41:34.1765061Z h5py/h5f.pyx:126: OSError
2021-11-26T20:41:34.1765500Z ________________ TestPageBuffering.test_only_with_page_strategy ________________
2021-11-26T20:41:34.1765793Z 
2021-11-26T20:41:34.1766199Z self = <h5py.tests.test_file.TestPageBuffering testMethod=test_only_with_page_strategy>
2021-11-26T20:41:34.1766622Z 
2021-11-26T20:41:34.1766991Z     def test_only_with_page_strategy(self):
2021-11-26T20:41:34.1767425Z         """Allow page buffering only with fs_strategy="page".
2021-11-26T20:41:34.1767821Z         """
2021-11-26T20:41:34.1768166Z         fname = self.mktemp()
2021-11-26T20:41:34.1768891Z >       with File(fname, mode='w', fs_strategy='page', page_buf_size=16*1024):
2021-11-26T20:41:34.1769237Z 
2021-11-26T20:41:34.1770215Z ../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.8/site-packages/h5py/tests/test_file.py:199: 
2021-11-26T20:41:34.1770996Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-11-26T20:41:34.1772071Z ../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.8/site-packages/h5py/_hl/files.py:507: in __init__
2021-11-26T20:41:34.1772840Z     fid = make_fid(name, mode, userblock_size, fapl, fcpl, swmr=swmr)
2021-11-26T20:41:34.1773923Z ../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.8/site-packages/h5py/_hl/files.py:226: in make_fid
2021-11-26T20:41:34.1774682Z     fid = h5f.create(name, h5f.ACC_TRUNC, fapl=fapl, fcpl=fcpl)
2021-11-26T20:41:34.1775130Z h5py/_objects.pyx:54: in h5py._objects.with_phil.wrapper
2021-11-26T20:41:34.1775514Z     ???
2021-11-26T20:41:34.1775885Z h5py/_objects.pyx:55: in h5py._objects.with_phil.wrapper
2021-11-26T20:41:34.1776265Z     ???
2021-11-26T20:41:34.1776659Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-11-26T20:41:34.1776967Z 
2021-11-26T20:41:34.1777247Z >   ???
2021-11-26T20:41:34.1777656Z E   OSError: Unable to create file (page buffering is disabled for parallel)
2021-11-26T20:41:34.1777952Z 
2021-11-26T20:41:34.1778280Z h5py/h5f.pyx:126: OSError
```

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -17,9 +17,6 @@ source:
   url: https://github.com/h5py/h5py/archive/{{ version }}.tar.gz
   sha256: 840fca0787bca8ad2729b9c20e06a16f219e7a24ccc59fdf5177a05ec8472b2d
   patches:
-    # override setup.py specified minimum compatible numpy versions
-    # we take care of those via numpy pinning below
-    # - adjust-minimum-numpy-run-versions.patch
     - osx-arm64.diff    # [osx and arm64]
     - osx-arm64-hdf5.diff    # [build_platform != target_platform]
 

--- a/recipe/run_test.py
+++ b/recipe/run_test.py
@@ -32,4 +32,7 @@ have_mpi = h5py.get_config().mpi
 assert have_mpi == should_have_mpi, "Expected mpi=%r, got %r" % (should_have_mpi, have_mpi)
 
 from sys import exit
-exit(h5py.run_tests())
+if have_mpi:
+    exit(h5py.run_tests("--with-mpi"))
+else:
+    exit(h5py.run_tests())

